### PR TITLE
mat64: add Copy to TriDense and drop Ler/Uer interfaces and methods

### PR DESCRIPTION
**Pull request by kortschak**
Depends #187.
Updates #138.

@btracey Please take a look.

### CODE REVIEW

File: mat64/triangular.go [hunk 1]
```diff
@@ -171,6 +171,63 @@ func (t *TriDense) Reset() {
...
+		row := make([]float64, c)
+		if t.isUpper() {
+			for i := 0; i < r; i++ {
+				a.Row(row, i)
+				copy(t.mat.Data[i*t.mat.Stride+i:i*t.mat.Stride+c], row[i:])
+			}
+		} else {
+			for i := 0; i < r; i++ {
+				a.Row(row, i)
+				copy(t.mat.Data[i*t.mat.Stride:i*t.mat.Stride+i+1], row[:c-i])
```
Comment (btracey) [inline mat64/triangular.go]: Shouldn't this be row[:i+1] ?

Comment (kortschak) [inline mat64/triangular.go]: Yes, it should.

**Comment by btracey** [comment]
LGTM

We should definitely add a fast past for TriDense - TriDense copying.

**Comment by kortschak** [comment]
What is the expected behavior u.Copy(l) in that case? I would expect just the diagonal.

**Comment by btracey** [comment]
Agreed.

**Comment by kortschak** [comment]
Fixed slow path - upper/lower logic was inverted. Nice to see our testing is tight.

**Comment by btracey** [comment]
Shoot! I thought I checked that, sorry. Yes, it is good.

**Comment by btracey** [comment]
It caught a bunch of bugs for me in the case of Dense multiplication where the underlying inputs are both *Dense. It's more complicated than one thinks at first.

**Pull request merged**